File: Mall-API/Mall-API/Mall-API/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Mall_API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Image { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "You must select a {0} from the list")]
        [Display(Name = "Category")]
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "You must select a {0} from the list")]
        [Display(Name = "Department")]
        public int DepartmentId { get; set; }
        public Department Department { get; set; }
    }
}

File: Mall-UI/src/app/models/product.ts
export interface Product {
    id: number
    name: string
    image: string
    categoryId: number
    categoryName: string
    departmentId: number
    departmentName: string
}

export interface MenuProduct {
    id: number
    name: string
}

export interface ProductCreationDTO {
    name: string
    image: File
    categoryId: number
    departmentId: number
}